packages/astro/src/env/runtime.ts

```diff
@@ -1,12 +1,15 @@
 import { AstroError, AstroErrorData } from '../core/errors/index.js';
 export { validateEnvVariable } from './validators.js';
+import { EventEmitter } from 'node:events'
```

Comment:
We probably shouldn't use this as it will not work in non-Node environments.

Since only our code should be using this anyways, can we just have it set a callback instead?

```js
export const onSetGetEnv = () => {};
```

```js
import {
	onSetGetEnv
} from 'astro/env/runtime';

onSetGetEnv = () => {
  // @@ON_SET_GET_ENV@@
};
```